src/Compilers/VisualBasic/Portable/Binding/Binder_ObjectInitializer.vb

```diff
@@ -508,6 +517,73 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Return resultExpression
         End Function
 
+        Private Shared Sub CheckRequiredMembersInObjectInitializer(
+            constructor As MethodSymbol,
+            containingType As NamedTypeSymbol,
+            initializers As ImmutableArray(Of BoundExpression),
+            creationSyntax As SyntaxNode,
+            diagnostics As BindingDiagnosticBag)
+
+            ' The only time constructor will be null is if we're trying to invoke a parameterless struct ctor, and it's not accessible (such as being protected).
+            Debug.Assert((constructor IsNot Nothing AndAlso ReferenceEquals(constructor.ContainingType, containingType)) OrElse containingType.IsStructureType())
+
+            If constructor IsNot Nothing AndAlso constructor.HasSetsRequiredMembers Then
+                Return
+            End If
+
+            If containingType.HasRequiredMembersError Then
+                ' A use-site diagnostic will be reported on the use, so we don't need to do any more checking here.
+                Return
+            End If
+
+            If TypeOf containingType Is SourceMemberContainerTypeSymbol Then
```

Comment:
>TypeOf containingType Is SourceMemberContainerTypeSymbol [](http://example.com/codeflow?start=15&length=56)

This condition won't work for constructed types and for retargeting scenario as well. I do no think suppression of the diagnostics is worth added complexity. I would get rid of this ```If```